diff --git a/docker/Dockerfile b/docker/Dockerfile
index 3afc63e..af9cab2 100644
--- a/docker/Dockerfile
+++ b/docker/Dockerfile
@@ -19,7 +19,8 @@ RUN  apt-get -y update && \
       libsuitesparse-dev \
       libglib2.0-dev \
       libyaml-cpp-dev \
-      libdw-dev
+      libdw-dev \
+      vim 
 
 #Install ROS
 # update ros repository
@@ -80,11 +81,17 @@ RUN   wget https://github.com/microsoft/onnxruntime/releases/download/v${ONNX_VE
       rm onnxruntime-linux-x64-gpu-${ONNX_VERSION}.tgz
 
 #Install Libtorch (CUDA)
-RUN   wget https://download.pytorch.org/libtorch/cpu/libtorch-shared-with-deps-${LIBTORCH_VERSION}%2Bcpu.zip && \
-      unzip libtorch-shared-with-deps-${LIBTORCH_VERSION}+cpu.zip && \
-      rm libtorch-shared-with-deps-${LIBTORCH_VERSION}+cpu.zip && \
+# RUN   wget https://download.pytorch.org/libtorch/cpu/libtorch-shared-with-deps-${LIBTORCH_VERSION}%2Bcpu.zip && \
+#       unzip libtorch-shared-with-deps-${LIBTORCH_VERSION}+cpu.zip && \
+#       rm libtorch-shared-with-deps-${LIBTORCH_VERSION}+cpu.zip && \
+#       cp -r libtorch/* /usr/local/
+
+RUN   wget https://download.pytorch.org/libtorch/cpu/libtorch-cxx11-abi-shared-with-deps-2.0.1%2Bcpu.zip && \
+      unzip libtorch-cxx11-abi-shared-with-deps-2.0.1+cpu.zip && \
+      rm libtorch-cxx11-abi-shared-with-deps-2.0.1+cpu.zip && \
       cp -r libtorch/* /usr/local/
 
+
 #Install LCM
 RUN   git clone https://github.com/lcm-proj/lcm && \
       cd lcm && \
@@ -159,8 +166,14 @@ RUN   mkdir -p ${SWARM_WS}/src/ && \
       git clone https://github.com/ros-perception/vision_opencv.git -b ${ROS_VERSION}
 COPY ./ ${SWARM_WS}/src/D2SLAM
 WORKDIR $SWARM_WS
-RUN   source "/opt/ros/${ROS_VERSION}/setup.bash" && \
+
+RUN   catkin config --extend /opt/ros/$ROS_VERSION && \
+      catkin config --merge-devel && \
       catkin config -DCMAKE_BUILD_TYPE=Release \
             --cmake-args -DONNXRUNTIME_LIB_DIR=/usr/local/lib \
-            -DONNXRUNTIME_INC_DIR=/usr/local/include && \
-      catkin build
+            -DONNXRUNTIME_INC_DIR=/usr/local/include 
+RUN apt-get update && apt-get install -y libspdlog-dev && \
+      rm -rf /var/lib/apt/lists/*
+RUN catkin build vision_opencv
+RUN catkin build d2frontend
+RUN catkin build
