diff --git a/cslam/loop_closure_detection_node.py b/cslam/loop_closure_detection_node.py
index d2980fa..597aede 100755
--- a/cslam/loop_closure_detection_node.py
+++ b/cslam/loop_closure_detection_node.py
@@ -19,16 +19,16 @@ class LoopClosureDetection(Node):
 
         self.declare_parameters(
             namespace='',
-            parameters=[('frontend.similarity_threshold', None),
-                        ('frontend.global_descriptor_technique', None),
-                        ('frontend.netvlad.pca_checkpoint', None), ('frontend.nn_checkpoint', None),
-                        ('robot_id', None), ('max_nb_robots', None),
+            parameters=[('frontend.similarity_threshold', rclpy.Parameter.Type.DOUBLE),
+                        ('frontend.global_descriptor_technique', rclpy.Parameter.Type.STRING),
+                        ('frontend.netvlad.pca_checkpoint', rclpy.Parameter.Type.STRING), ('frontend.nn_checkpoint', rclpy.Parameter.Type.STRING),
+                        ('robot_id', rclpy.Parameter.Type.INTEGER), ('max_nb_robots', rclpy.Parameter.Type.INTEGER),
                         ('frontend.inter_robot_loop_closure_budget', 5),
                         ('frontend.inter_robot_detection_period_sec', 5),
-                        ('frontend.nb_best_matches', 10), ('frontend.image_crop_size', None),
+                        ('frontend.nb_best_matches', 10), ('frontend.image_crop_size', rclpy.Parameter.Type.INTEGER),
                         ('frontend.intra_loop_min_inbetween_keyframes', 10),
-                        ('neighbor_management.max_heartbeat_delay_sec', 5),
-                        ('neighbor_management.init_delay_sec', 5),
+                        ('neighbor_management.max_heartbeat_delay_sec', 5.0),
+                        ('neighbor_management.init_delay_sec', 5.0),
                         ('neighbor_management.heartbeat_period_sec', 0.5),
                         ('frontend.detection_publication_period_sec', 1.0),
                         ('frontend.detection_publication_max_elems_per_msg',
