diff --git a/src/back_end/decentralized_pgo.cpp b/src/back_end/decentralized_pgo.cpp
index da5fe2c..acd80fc 100644
--- a/src/back_end/decentralized_pgo.cpp
+++ b/src/back_end/decentralized_pgo.cpp
@@ -536,7 +536,7 @@ bool DecentralizedPGO::check_waiting_timeout()
   {
     end_waiting();
     optimizer_state_ = OptimizerState::IDLE;
-    RCLCPP_INFO(node_->get_logger(), "Timeout: " + std::to_string(robot_id_));
+    RCLCPP_INFO(node_->get_logger(), ("Timeout: " + std::to_string(robot_id_)).c_str());
   }
   return is_waiting();
 }
diff --git a/src/back_end/utils/simulated_rendezvous.cpp b/src/back_end/utils/simulated_rendezvous.cpp
index 05de85f..a81d3ae 100644
--- a/src/back_end/utils/simulated_rendezvous.cpp
+++ b/src/back_end/utils/simulated_rendezvous.cpp
@@ -17,7 +17,7 @@ SimulatedRendezVous::SimulatedRendezVous(std::shared_ptr<rclcpp::Node> &node,
                 auto delim0 = line.find(",");
                 if (robot_id_ == std::stoul(line.substr(0, delim0)))
                 {
-                    RCLCPP_INFO(node_->get_logger(), "Simulated rendezvous schedule of robot " + line);
+                    RCLCPP_INFO(node_->get_logger(), ("Simulated rendezvous schedule of robot " + line).c_str());
                     while (delim0 != std::string::npos)
                     {
                         auto delim1 = line.find(",", delim0 + 1);
